core/src/main/scala/stryker4s/config/implicits/ConfigReaderImplicits.scala

```diff
@@ -27,18 +28,25 @@ trait ConfigReaderImplicits extends Logging {
     })
 
   private[config] implicit val exclusions: ConfigReader[Exclusions] =
-    ConfigReader[List[String]]
-      .map(warnInvalidExclusions)
+    ConfigReader.fromCursor[List[String]](errorOnInvalidExclusions)
       .map(exclusions => Exclusions(exclusions.toSet))
 
-  private def warnInvalidExclusions(list: List[String]): List[String] = {
-    val (valid, invalid) = list.partition(Mutation.mutations.contains)
-    if (invalid.nonEmpty) {
-      val errorMessage =
-        s"""Invalid exclusion option(s): '${invalid.mkString(", ")}'
-            |Valid exclusions are ${Mutation.mutations.mkString(", ")}""".stripMargin
-      throw new Exception(errorMessage)
-    }
-    valid
+  private def errorOnInvalidExclusions(configCursor: ConfigCursor): Either[ConfigReaderFailures, List[String]] = {
```

Comment:
Here's my attempt at making it more in line with PureConfig. I might need help for this one. :/

Reply:
pff, this added a lot more complexity then I expected. I suspected `FailureReason` would extend exception in some sort and then it would be really easy to create a custom exception that extended from FailureReason.

Sorry for this but I think the first solution is a bit more understandable. You could still make a custom exception in the exception package. An example can be found here: https://github.com/stryker-mutator/stryker4s/blob/master/util/src/main/scala/stryker4s/extensions/exceptions/UnableToBuildPatternMatchException.scala

Reply:
Yeah sadly `FailureReason` isn't throwable. You have to return it in an Either monad which makes the whole thing quite complex. 😅 
I agree. So I changed it to a custom exception now. 👍